packages/@expo/config-plugins/src/plugins/mod-compiler.ts

```diff
@@ -103,13 +107,15 @@ function getRawClone({ mods, ...config }: ExportedConfig) {
   return Object.freeze(JSON.parse(JSON.stringify(config)));
 }
 
-const orders: Record<string, string[]> = {
-  ios: [
+const precedences: Record<string, Record<string, number>> = {
+  ios: {
     // dangerous runs first
-    'dangerous',
+    dangerous: 2,
     // run the XcodeProject mod second because many plugins attempt to read from it.
-    'xcodeproj',
-  ],
+    xcodeproj: 1,
+    // patch at the last
+    patch: -1,
```

Comment:
Change this to something like `finalize` or `post` to indicate that it runs last. Split this feature out of this PR and land it as a new dangerous modifier that runs after all other modifiers.

```suggestion
    finalize: -1,
```